File: CheckIn.Api.Common/Models/Lists/CheckInEventListModel.cs
using CheckIn.Api.Common.Models.Interfaces;

namespace CheckIn.Api.Common.Models.Lists;

public class CheckInEventListModel : IEntityModel
{
    public Guid Id { get; set; }
    public required string Title { get; set; }
    public required string Hash { get; set; }
    public DateTime CreatedAt { get; set; }
    public Guid OwnerId { get; set; }
}

File: checkin_frontend/lib/models/checkin_event/checkin_event_list_model.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'checkin_event_list_model.freezed.dart';

part 'checkin_event_list_model.g.dart';

@freezed
abstract class CheckInEventListModel with _$CheckInEventListModel {
  const factory CheckInEventListModel({
    required String id,
    required String title,
    required String hash,      
    required DateTime createdAt,
    required String ownerId,
  }) = _CheckInEventListModel;

  factory CheckInEventListModel.fromJson(Map<String, dynamic> json) =>
      _$CheckInEventListModelFromJson(json);
}
